src/TLuaInterpreter.cpp

```diff
@@ -5724,6 +5798,197 @@ int TLuaInterpreter::loadSoundFile(lua_State* L)
+                stringValue.replace('/', R"(\)");
+            } else {
+                stringValue.replace('\\', "/");
+            }
+
+            mediaData.setMediaFileName(stringValue);
+            break;
+        case 2:
+            intValue = getVerifiedInt(L, __func__, i, "volume");
+
+            if (intValue == TMediaData::MediaVolumePreload) {
+                {
+                } // Volume of 0 supports preloading
+            } else if (intValue > TMediaData::MediaVolumeMax) {
+                intValue = TMediaData::MediaVolumeMax;
+            } else if (intValue < TMediaData::MediaVolumeMin) {
+                intValue = TMediaData::MediaVolumeMin;
+            }
+
+            mediaData.setMediaVolume(intValue);
+            break;
+        case 3:
+            intValue = getVerifiedInt(L, __func__, i, "start");
+
+            if (intValue < 0) {
+                lua_pushfstring(L, "playVideoFile: bad argument range for %s (values must be greater than or equal to 0, got value: %d)", "start", intValue);
+                return lua_error(L);
+            }
+
+            mediaData.setMediaStart(intValue);
+            break;
+        case 4:
+            intValue = getVerifiedInt(L, __func__, i, "loops");
+
+            if (intValue < TMediaData::MediaLoopsRepeat || intValue == 0) {
+                intValue = TMediaData::MediaLoopsDefault;
+            }
+
+            mediaData.setMediaLoops(intValue);
+            break;
+        case 5:
+            stringValue = getVerifiedString(L, __func__, i, "key");
+            mediaData.setMediaKey(stringValue);
+            break;
+        case 6:
+            stringValue = getVerifiedString(L, __func__, i, "tag");
+            mediaData.setMediaTag(stringValue);
+            break;
+        case 7:
+            boolValue = getVerifiedBool(L, __func__, i, "continue");
+            mediaData.setMediaContinue(boolValue);
+            break;
+        case 8:
+            stringValue = getVerifiedString(L, __func__, i, "url");
+            mediaData.setMediaUrl(stringValue);
+            break;
+        }
+    }
+
+    if (mediaData.getMediaFileName().isEmpty()) {
+        return warnArgumentValue(L, __func__, QLatin1String("missing argument 1 (file to play)"));
+    }
+
+    mediaData.setMediaProtocol(TMediaData::MediaProtocolAPI);
+    mediaData.setMediaType(TMediaData::MediaTypeVideo);
+    host.mpMedia->playMedia(mediaData);
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// Private
+int TLuaInterpreter::playVideoFileAsTableArgument(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    TMediaData mediaData{};
+
+    lua_pushnil(L);
+    while (lua_next(L, 1) != 0) {
+        // key at index -2 and value at index -1
+        QString key = getVerifiedString(L, __func__, -2, "table keys");
+        key = key.toLower();
+
+        if (key == QLatin1String("name") || key == QLatin1String("url") || key == QLatin1String("key") || key == QLatin1String("tag")) {
```

Comment:
OTOH You can avoid the lower-casing entirely (i.e. don't apply the above suggestion) and use `QString::compare(...)` in its case-insensitive mode - see "30. qstring-insensitive-allocation" in **["Uncovering 32 Qt best-practices with compile-time Clazy"](https://www.kdab.com/uncovering-32-qt-best-practices-compile-time-clazy/)**:
```suggestion
        if (!key.compare(QLatin1String("name"), Qt::CaseInsensitive) || (!key.compare(QLatin1String("url"), Qt::CaseInsensitive) || (!key.compare(QLatin1String("key"), Qt::CaseInsensitive) || (!key.compare(QLatin1String("tag"), Qt::CaseInsensitive)) {
```